doc/catalog-format-spec.md

```diff
@@ -508,6 +508,63 @@ static, i.e. they cannot depend on the context.
 
+## Work-in-progress dependency overrides
+
+It is usual to develop several interdependent crates at the same time. In this scenario, it is often impractical to rely on indexed releases which are not intended to be modified. Instead, one would prefer to use a work-in-progress version of a crate to fulfill some dependency.
+
+Alire provides *pins* to support this use case. Pins are overrides to dependencies, are intended to be used locally, and to be fulfilled by proper dependencies once a crate is ready to be published. The use of pins is based on two ideas:
+
+* Dependencies are given, as normally, in the `depends-on` array of the manifest, even for those dependencies to be pinned. This way, once the release is ready, pins are simply removed and the actual dependencies are used in their place.
+* Dependency overrides, aka *pins*, are given under the `[[pins]]` array of the manifest.
+
+Three kinds of pins are available, all of them with the syntax:
+
+`crate_name = { pin_attributes }`
+
+The specific pin kinds and their attributes are:
+
+* Pins to versions: used to force the use of a particular version of an indexed crate.
+
+  * `version`: a string containing a single version to be used.
+  * `crate_name = { version = "1.2+hotfix-1" }`
+
+* Pins to local crates: a local directory will fulfill the crate dependency, no matter what version is given in its local manifest. "Raw" Ada projects without an Alire manifest can be used too, as long as their project file matches the crate name and it is located in the directory given as override.
+
+  * `path`: an absolute or relative path to the crate directory.
+  * `crate_name  = { path = "../my/wip/crate" }`
+
+  For the common case of directories containing an Alire manifest, dependencies and pins will be included recursively in the build context.
+
+* Pins to git repositories: the repository will be cloned locally and its directory will be used as in the previous case. Currently, this pin may optionally include a commit to fix the checkout to be used. Otherwise, the default branch will be used, and running `alr update` will refresh the checkout.
+
+  * `url`: the URL of a git repository
+  * `commit` (optional): a complete git commit hash.
+  * `crate_name = { url = "https://my/repo.git" } # Updatable pin`
+  * `crate_name = { url = "https://my/repo.git", commit="abcdef..." } # Fixed pin`
+
+### Using pins for crate testing
+
+Pins are also useful to have a separate test project that depends on your main crate. The recommended setup is as follows:
+
+```
+/path/to/my_crate
+├── alire.toml
+└── my_crate_test
+    └── alire.toml
+```
+
+I.e., a `my_crate_test` crate is initialized within the main `my_crate`. In the `my_crate_test` manifest, you have a dependency and local relative path pin for `my_crate`:
+
+```toml
+# my_crate_test/alire.toml
+[[depends-on]]
+my_crate = "*"              # Any version of the main crate
+[[pins]]
+my_crate = { path = ".." }  # Overridden by the latest sources
+```
+
```

Comment:
You can mention dependencies that are specific for testing.

There is also a the fact that you can set crate config variables specifically for testing, but that's maybe too complex here.